### PR TITLE
Piotr wojdas lista 3

### PR DESCRIPTION
**Pull request by Piotr-Wojdas**
Wiem, że można byłoby się pobawić jeszcze kanałami modelu i innymi parametrami ale tymczasowy brak gpu i limity czasowe (dosłownie jak i collaba) sprowadziły mnie do zrobienia tylko tego. Jak dorwę się w końcu do peceta to będzie można gotować

### CODE REVIEW

**Review by maxster256 — COMMENTED** [review]
Opisy poszczególnych etapów zadania (przygotowanie danych, opisy poszczególnych funkcji oraz opis trenowania modeli) ułatwiają zapoznanie się z treścią notatnika. Podoba mi się również wybór hiperparametrów do zbadania, czyli `learning_rate` i `optimizer`. Cała praca myślę, że jest naprawdę dobrze wykonana i nie ma za bardzo do czego się przyczepić :) <br>
Na następny raz rozważyłbym tylko dopracowanie trochę wizualizacji. Zamiast tworzyć kilka dużych plotów to można byłoby stworzyć jeden plot z kilkoma subplotami, opcjonalnie też można skrócić kod odpowiedzialny za tworzenie tych wizualizacji do jednej funkcji wywoływanej kilka razy, tylko z innymi parametrami, no i fajnie byłoby dodać siatkę `plt.grid()` do wykresów liniowych dla łatwiejszego odczytywania wartości z tych wykresów.

**Review by Sasserine — COMMENTED** [review]
Super praca, dobre komentarze, trafne wnioski, spoko analiza. Nie sądzę żebym miała co poprawiać

**Comment by woznin** [comment]
Świetna praca!
Mega plusy za dostrojenie hiperparametrów, testy na różnych wersjach modelu i dokładne podsumowanie na czym się myli itd.
W przyszłości można to troszke przyspieszyć frameworkiem [Optuna](https://optuna.readthedocs.io/en/stable/tutorial), który wykorzystuje optymalizacje Bayesowską do tego samego co GridSearch ale nie wykonuje przeglądu zupełnego.
Naprawdę dobra robota.

Taka tyci notka na przyszłość - do testów learning rate'u. Jak wiemy ze w 10 epok uzyskalismy +- optimum na lr = 0.01, to dla lr mniejszych niż 0.01 osiągnięcie optimum może zająć dłużej (skracamy krok wzdłuż spadku gradientu, więc dojście minimum może zająć dłużej).
